test/Kestrel.InMemory.FunctionalTests/HttpProtocolSelectionTests.cs

```diff
@@ -35,11 +35,34 @@ public Task Server_Http1Only_Cleartext_Success()
         }
 
         [Fact]
-        public Task Server_Http2Only_Cleartext_Success()
+        public async Task Server_Http2Only_Cleartext_Success()
         {
             // Expect a SETTINGS frame (type 0x4) with default settings
-            return TestSuccess(HttpProtocols.Http2, Encoding.ASCII.GetString(Http2Connection.ClientPreface),
-                "\x00\x00\x06\x04\x00\x00\x00\x00\x00\x00\x03\x00\x00\x00\x64");
+            var expected = new byte[]
+            {
+                0x00, 0x00, 0x0C, // Payload Length (6 * settings count)
+                0x04, 0x00, 0x00, 0x00, 0x00, 0x00, // SETTINGS frame (type 0x04)
+                0x00, 0x03, 0x00, 0x00, 0x00, 0x64, // Connection limit
+                0x00, 0x06, 0x00, 0x00, 0x80, 0x00 // Header size limit
```

Comment:
Any particular reason for adding the header size limit to the settings frame here?

Reply:
Kestrel sends all non-protocol-default settings at the start of every connection. This output is used here to confirm kestrel is treating the connection as HTTP/2.

Reply:
Right. I had this confused with Http2Connection.ClientPreface.

Reply:
It doesn't make much sense for the *client* to send these settings, but I didn't really think it through fully.